modules/models/src/main/java/org/eclipse/xpanse/modules/models/security/model/CurrentUserInfoCacheHolder.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.modules.models.security.model;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.springframework.context.annotation.Profile;
+
+/**
+ * User information cache.
+ */
+@Profile("zitadel")
+public class CurrentUserInfoCacheHolder {
+
+    public static final String CURRENT_USERINFO_CACHE_NAME = "CurrentUserinfo";
+
+    private static final Map<String, CurrentUserInfo> USERINFO_CACHE
```

Comment:
Using this solution, the map seems always store only the latest putted user . Please test if the user information obtained in the sub thread is correct if multiple users access it simultaneously. If not correct, you can try to use the `ThreadLocal<CurrentUserInfo>` to obtain the user information in sub threads.